builtins/amp-video.js

```diff
@@ -36,7 +36,31 @@ export function installVideo(win) {
     }
 
     /** @override */
+    prerenderAllowed() {
+      return true;
+    }
+
+    /** @override */
+    buildCallback() {
+      /** @private */
+      this.realPreloadAttrValue_ = this.element.getAttribute('preload');
+      const posterAttr = this.element.getAttribute('poster');
+
+      if (!posterAttr) {
+        console/*OK*/.warn(
+            'No "poster" attribute has been provided for amp-video.');
+      }
+    }
+
+    /** @override */
     layoutCallback() {
+      const docState = documentStateFor(this.getWin());
+      // Disable video preload in prerender mode and reset it when rendered.
+      if (docState.getVisibilityState() === 'prerender') {
```

Comment:
It's a possible state where the document is "visible" as far as the browser concerned, but viewer haven't yet shown it to the user (e.g. CSS `visibility: hidden`). So, the question is whether we want to wait for preload until the document is actually shown to the user.